nucliadb/src/nucliadb/ingest/orm/knowledgebox.py

```diff
@@ -463,6 +486,54 @@ async def delete_vectorset(self, vectorset_id: str):
         shard_manager = get_shard_manager()
         await shard_manager.delete_vectorset(self.kbid, vectorset_id)
 
+    @classmethod
+    async def _maybe_create_external_index(
+        cls,
```

Comment:
If you're in into having nucliadb index node as an index provider, we could maybe work on the `KBShardManager` abstraction to become the nucliadb index provider and have this pinecone logic there, outside the `KnowledgeBox` class